pkg/signup/signup_service_test.go

```diff
@@ -286,14 +367,16 @@ func (s *TestSignupServiceSuite) TestGetSignupUnknownStatus() {
 	require.EqualError(s.T(), err, "unable to parse readiness status as bool: blah-blah-blah: strconv.ParseBool: parsing \"blah-blah-blah\": invalid syntax")
 }
 
-func newSignupServiceWithFakeClient() (signup.Service, *fake.FakeUserSignupClient, *fake.FakeMasterUserRecordClient) {
+func newSignupServiceWithFakeClient() (signup.Service, *fake.FakeUserSignupClient, *fake.FakeMasterUserRecordClient, *fake.FakeBannedUserClient) {
 	fakeClient := fake.NewFakeUserSignupClient(TestNamespace)
 	fakeMURClient := fake.NewFakeMasterUserRecordClient(TestNamespace)
+	fakeBannedUserClient := fake.NewFakeBannedUserClient(TestNamespace)
 	return &signup.ServiceImpl{
 		Namespace:         TestNamespace,
 		UserSignups:       fakeClient,
 		MasterUserRecords: fakeMURClient,
-	}, fakeClient, fakeMURClient
+		BannedUsers:       fakeBannedUserClient,
+	}, fakeClient, fakeMURClient, fakeBannedUserClient
```

Comment:
I'm slightly missing the purpose of these fake clients - I believe that they could be replaced by the usage of `fake.RESTClient` and `fake.FakeDynamicClient` or am I missing something?

Reply:
These were added a long time ago based on a tutorial for implementing custom k8s clients for CRDs via the RESTClient API.  I'm fine with replacing them with something else if there's a compelling reason to do so but this would be outside the scope of this PR.

Reply:
But you created a new `FakeBannedUserClient` so my comment wasn't related only to the old ones that were using RESTClient but mainly to the new one that is using `DynamicClient`. Could you please tell me what is the reason behind that and why you don't want to use the proper fake client implementation?